zcash_client_sqlite/src/wallet/sapling.rs

```diff
@@ -82,9 +88,19 @@ impl ReceivedSaplingOutput for DecryptedOutput<Note> {
     fn note_commitment_tree_position(&self) -> Option<Position> {
         None
     }
+    fn recipient_key_scope(&self) -> Scope {
+        if self.transfer_type == TransferType::WalletInternal {
```

Comment:
There was a prior issue where `TransferType::WalletInternal` does not directly correlate with either `is_change` or the decryption key scope. Now that this PR is introducing a migration to explicitly store the recipient key scope, how is the way `TransferType::WalletInternal` gets handled being altered to fix the prior issue?

Reply:
I checked that all of the places that `TransferType::WalletInternal` can arise are, at least at present, a consequence of funds being sent to the internal-scoped IVK. So I think that this is fine.

`is_change` might be false for shielding transactions while still being wallet-internal. But that's fine.

Reply:
What about for historic change outputs created before the introduction of internal-scoped keys? Are these correctly detected as `is_change = true`?

Reply:
In Slack, @nuttycom told me that there is no change to the `is_change` problem for older wallets. This might be what #936 is tracking.